src/Microsoft.AspNet.Server.Kestrel/Http/Frame.Generated.cs

```diff
@@ -0,0 +1,297 @@
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http 
+{
+    public partial class Frame
+    {
+        
+        private object _currentIHttpRequestFeature;
```

Comment:
Because they're only `[]` and `[]=`ed as object. Downcasting at this level could make the behavior different compared to `Dictionary<Type,Object>` on one hand, and would add a tiny bit of cpu on assignment that will be repeated anyway when a caller that asks for the feature downcasts again anyway.